crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -125,6 +125,22 @@ impl UserIdentity {
         }
     }
 
+    /// Consider this identity to be trusted.
+    ///
+    /// Resets this identity to the same "pinned" state assigned the first time
+    /// a user is encountered, where we follow "trust on first use" and
+    /// consider it trusted.
```

Comment:
I must admit I don't understand this explanation at all.

The first thing that throws me is the word "trusted". Do we still retain the ability to differentiate between identities which are simply pinned versus properly verified?

Furthermore, the word "resets" implies to me that we will force the "pinned" state, replacing any previous state, even if it was "verified". Is this the case?

Reply:
Yes, this is very confusing at the moment! How about this wording:

```
/// "Pin" this identity, meaning it is not in "pin violation".
///
/// When we first see a user, we assume their cryptographic identity is correct.
/// We call this "pinning" the identity. If they later change their identity, then they are
/// in "pin violation": we require user interaction before they are pinned again.
///
/// Note: this is not related to verification of users, which is a more secure way of
/// confirming that the identity you are communicating with is really the person you
/// think it is.
```

Reply:
(Applied in 2d89f6cab4 but definitely up for further discussion)

Reply:
```rust
/// Remember this identity, ensuring it does not result in a pin violation.
///
/// When we first see a user, we assume their cryptographic identity has not
/// been tampered with by the homeserver or another entity with
/// man-in-the-middle capabilities. We remember this identity and call this
/// action "pinning".
///
/// If the identity presented for the user changes later on, the newly presented
/// identity is considered to be in "pin violation". This method explicitly
/// accepts the new identity, allowing it to replace the previously pinned one
/// and bringing it out of pin violation.
///
/// An identity which is not verified and is in pin violation will require user
/// approval before it can be used. See
/// [`OtherUserIdentity::identity_needs_user_approval`].
```

How about this instead?

I aimed to avoid glossing over verification as an entirely separate matter, instead linking pinning and verification to show how they jointly impact the system. I also spell out what "correct" means when talking about an identity.